### ISSUE TITLE
Opening the Car Thing

### ISSUE THREAD

**Issue by big-harry** [err4o4/spotify-car-thing-reverse-engineering#14]
At a temperature of roughly 180F (80C) I was able to melt the glue at remove the screen. I do not recommend holding the Car Thing while melting the glue.

![image](https://user-images.githubusercontent.com/94585779/196294206-0c0cf823-4676-4e8c-a5f7-4d858fd90043.jpeg)

![image](https://user-images.githubusercontent.com/94585779/196294253-bd683477-120c-432d-b4d5-f21a120cfc5b.jpeg)

**Comment by big-harry** [comment]
![image](https://user-images.githubusercontent.com/94585779/196294667-40f3868e-0bf4-4cf3-a6fd-36bcc214146d.jpeg)

**Comment by big-harry** [comment]
When opening the unit, there's a brick connector which may come off when opening. On the inside, there's 3 screws which have a six sided star shape on top.

**Comment by big-harry** [comment]
They are T6 screws. Removing them and disconnecting the ribbon cable is all that's left to fully tear down the Car Thing.

**Comment by big-harry** [comment]
![Front](https://user-images.githubusercontent.com/94585779/196303642-6f628301-f8f1-46b8-9069-6323efc46a78.jpg)

I measured most of the test pads, one of them is 38V and that's not a mistake.

**Comment by filipcarlsson** [comment]
If you can figure out where the flash is maybe you can use glitching by shorting a pin on the flash to ground if possible somehow to corrupt u-boot process? Just a thought.

**Comment by big-harry** [comment]
<img width="707" src="https://user-images.githubusercontent.com/94585779/196716942-448b5bd4-ba8e-47cc-b7bc-02007669d0f2.png">

Looking at the part, it would be hard to short it because it's BGA.

<img width="332" src="https://user-images.githubusercontent.com/94585779/196717935-70ebd866-bd06-4fce-9d2e-e2c7a37689bd.png">

I'm guessing I might be able to scrape the masking off to short it, but it might break it. Also I don't really know what we can do with a corrupted boot process, I think we should see which pins correspond to UART.

**Comment by filipcarlsson** [comment]
Usually it would kick you to a u-boot shell. Not really sure about this one tho.

**Comment by williamtcastro** [comment]
Are we able to find the flash memory capacity ?

**Comment by justMaku** [comment]
> Are we able to find the flash memory capacity ?

Most likely 4Gb - 2gb data, 2x512mb for os partitions, last 1gb is some other random system partitions.

**Comment by williamtcastro** [comment]
> > Are we able to find the flash memory capacity ?
> 
> Most likely 4Gb - 2gb data, 2x512mb for os partitions, last 1gb is some other random system partitions.

I've looked up for the datasheets of the components and the EMMC its in fact 4GB and surprisingly enough the car thing as a 4Gb ram chip  

[4gb_ddr3_e_die_component_datasheet.pdf](https://github.com/err4o4/spotify-car-thing-reverse-engineering/files/9879881/4gb_ddr3_e_die_component_datasheet.pdf)

**Comment by Merlin04** [comment]
Note that 4Gb=4 gigabits, which is equivalent to 512 megabytes

**Comment by williamtcastro** [comment]
> Note that 4Gb=4 gigabits, which is equivalent to 512 megabytes

Yes that was pointed out for me on reddit, thanks, sorry for the rookie mistake
